src/block-data-providers/esplora.provider.ts

```diff
@@ -161,26 +164,27 @@ export class EsploraProvider
     private async processBlock(height: number, hash: string) {
         const txids = await this.getTxidsForBlock(hash);
 
-        // this is not very efficient
-        // at the time of implementation, I'm using a public esplora instance,
-        // so there is a rate limit on the number of requests
-        // in the future, we should add a flag to parallelize this
-        for (let i = 1; i < txids.length; i++) {
-            const txid = txids[i];
-            const tx = await this.getTx(txid);
-            const vin: TransactionInput[] = tx.vin.map((input) => ({
-                txid: input.txid,
-                vout: input.vout,
-                scriptSig: input.scriptsig,
-                prevOutScript: input.prevout.scriptpubkey,
-                witness: input.witness,
-            }));
-            const vout = tx.vout.map((output) => ({
-                scriptPubKey: output.scriptpubkey,
-                value: output.value,
-            }));
-
-            await this.indexTransaction(txid, vin, vout, height, hash);
+        for (let i = 1; i < txids.length; i += this.batchSize) {
+            const batch = txids.slice(i, i + this.batchSize);
```

Comment:
slice is a costly operation, I bet there is a way to remove it.
Think of some more performance optimisations that you can do, like removing `map` etc

Reply:
should i use a different for loop for slice operation?

Reply:
nvm this is fine